### DISCUSSION TITLE
Building and running homepage example?

### DISCUSSION THREAD

**Discussion by g40** [picoe/Eto#2448] [Q&A] [answered]
Hello.

Using VS2022 on Windows 10 and Mac with a fresh clone of the Eto repository. Questions arising are:

1. How do I get a build?
2. Is there a VS project/solution to build the application shown here? https://github.com/picoe/Eto/blob/develop/images/windows.png
3. Is the process the same on a Mac?

Many thanks.

**Comment by g40** [comment]
After cloning, make sure submodules are present. Otherwise MonoMac will get omitted. 

In my case I cloned using `git clone --recurse-submodules git@github.com:g40/Eto.git`

1. To build using VS2022, ensure .NET framework 4.6.1 is installed using the VS installer. 
3. Then open `Eto\src\Eto.sln`
4. Make sure the correct startup project is selected in the VS toolbar. In this case it's `Debug+Windows+Eto.Test.WinForms`
5. Hit F5 to build and run.
6. Is the process the same on a Mac? Not yet known as my elderly Intel Mini will not run the latest VS for Mac.

Confusingly the `dotnet` tool does not show a complete listing:

```
dotnet --list-sdks
7.0.202 [C:\Program Files\dotnet\sdk]

dotnet --list-runtimes
Microsoft.AspNetCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]```

*N.B.* To build and run the WPF variety you also need to install .NET framework 4.6.2 (!)

**Comment by cwensley** [comment, marked as answer]
Hey @g40,

On Windows, open Eto.sln in VS 2022 and set the startup project as you mentioned.

On Mac and Linux, open the folder in VS Code, install the C# plugin, and set the debug target to one of the appropriate test projects like Eto.Test.Mac64, Eto.Test.Gtk, etc and press the play button.

You can also use VS Code on Windows.

I don't support using Visual Studio on Mac to develop Eto at this point.

Hope this helps!

**Comment by g40** [comment]
@cwensley Thank you. Great work as well.
